src/utils/nanoContracts.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import hathorLib from '@hathor/wallet-lib';
+
+/**
+ * Methods to handle Nano Contracts
+ *
+ * @namespace Nano Contracts
+ */
+
+const nanoContracts = {
+  /**
+   * Get registered nano contracts from the wallet instance.
+   * @param {HathorWallet} wallet
+   * @returns {Promise<Record<string, NanoContractData>>}
+   */
+  async getRegisteredNanoContracts(wallet) {
+    // redux-saga generator magic does not work well with the "for await..of" syntax
+    // The asyncGenerator is not recognized as an iterable and it throws an exception
+    // So we must iterate manually, awaiting each "next" call
+    const iterator = wallet.storage.getRegisteredNanoContracts();
+    const nanoContracts = {};
+    let next = await iterator.next();
+    while (!next.done) {
+      const ncData = next.value;
+      nanoContracts[ncData.ncId] = { ...ncData, history: [], historyMetadata: {} };
+      next = await iterator.next();
+    }
+
+    return nanoContracts;
+  },
+
+  /**
+   * Check if transaction is a nano contract create tx
+   *
+   * @param {Transaction} tx
+   * @returns {boolean}
+   */
+  isNanoContractCreate(tx) {
+    return tx.version === hathorLib.constants.NANO_CONTRACTS_VERSION && tx.nc_method === hathorLib.constants.NANO_CONTRACTS_INITIALIZE_METHOD;
+  },
```

Comment:
This makes more sense as an util from the lib, what do you think?

Reply:
The method was added to the lib: https://github.com/HathorNetwork/hathor-wallet-lib/pull/691

As soon as the version is released I will create a PR to use it here. (if this PR takes more time to be merged, I will use it here already)